### PR TITLE
[TTAHUB-2317] AR groups

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

When creating a recipient AR we now allow the user to select a group that will pre fill the list of recipients. If the pre filled list of recipients is altered in anyway (add/remove). The user is informed that they have modified the group recipients and will be given the option to start over.

**NOTE 12/07/2023:** Right now we are getting all groups by region regardless of shared users as this change is still pending on the BE. **ATM the FE can be reviewed.**

## How to test

- Review all code.
- Create a recipient report and test the workflow from the UX Workflow:
 https://www.figma.com/proto/zciEyTLd2MeSK7z5YVKdbF/Create-AR-and-TR-session-from-%22my-group%22?page-id=1%3A52&type=design&node-id=531-7254&viewport=-848%2C-3704%2C0.25&t=GBVp5tULq3N2QVO3-1&scaling=min-zoom&starting-point-node-id=531%3A7254&show-proto-sidebar=1&mode=design*
- Test validation for groups.
- Make sure the group displayed are correct for the region and number of recipients populated.
- Test edge cases of selecting and removing groups also having pre selected recipients and switching (and vice versa).
- Make sure a report can be created and approved successfully when using and not using groups.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2317

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/activitySummary.js [hunk 1]
```diff
@@ -151,29 +253,63 @@ const ActivitySummary = ({
...
+                  control={control}
+                  id="recipientGroup"
+                  name="recipientGroup"
+                  inputRef={register({ required: 'Select a group' })}
+                  onAbort={resetGroup}
+                  onChange={handleGroupChange}
+                >
+                  <option value="" disabled selected hidden>- Select -</option>
+                  {groups.map((group) => (
+                    <option key={group.id} data-key={group.id}>{group.name}</option>
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/activitySummary.js:287]: Thanks, wasn't sure the best way to handle this case.

File: frontend/src/pages/ActivityReport/Pages/activitySummary.js [hunk 2]
```diff
@@ -94,6 +124,61 @@ const ActivitySummary = ({
...
+    if (useGroup
+      && watchGroup
+      && (groupRecipientIds.length !== watchFormRecipients.length
+        || selectedRecipientsNotInGroup.length > 0)) {
+      setShowGroupInfo(true);
+      setUseGroup(false);
+      setValue('recipientGroup', null, { shouldValidate: false });
+      setGroupRecipientIds([]);
+    }
+  }, [watchFormRecipients]);
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/activitySummary.js:164]: Odd that this wasn't flagged by the linter?

**Review by thewatermethod — APPROVED** [review]
Nice!

Should "Group" be capitalized here?

Screenshot 2023-12-08 at 2 20 30 PM

**Review by nvms — COMMENTED** [review]
Would like to handle the case of "Use groups" being selected, but the user has no groups. Currently an empty Select is shown, and nothing happens when you click on an empty Select, which feels weird.

Like we talked about, maybe we introduce logic to just not show the "Use groups" checkbox when there are no groups to display.

**Comment by AdamAdHocTeam** [comment]
> Would like to handle the case of "Use groups" being selected, but the user has no groups. Currently an empty Select is shown, and nothing happens when you click on an empty Select, which feels weird.
> 
> Like we talked about, maybe we introduce logic to just not show the "Use groups" checkbox when there are no groups to display.

This has been fixed to hide the check box if there are no groups.

**Review by nvms — APPROVED** [review]
